src/compressed_tensors/utils/offload.py

```diff
@@ -161,25 +161,32 @@ def register_offload_parameter(
     module: torch.nn.Module,
     name: str,
     parameter: torch.nn.Parameter,
+    offload_device: Optional[Union[torch.device, Literal["disk"]]] = None,
 ):
     """
     Register a parameter to the given module which may be offloaded
 
     :param module: maybe offloaded module
     :param name: name of newly registered parameter
     :param parameter: parameter being registered
+    :param offload_device: device on which weight will be offloaded to. If None is
+        provided, then infer device from parameters on module
     """
+    has_onload = any(p.device != torch.device("meta") for p in module.parameters())
     module.register_parameter(name, parameter)
 
     if has_offloaded_params(module):
-        update_offload_parameter(module, name, parameter.data)
-        set_module_tensor_to_device(module, name, "meta")
+        weights_map = module._hf_hook.weights_map
+        offload_to_weights_map(weights_map, name, parameter.data, offload_device)
+        if not has_onload:
+            set_module_tensor_to_device(module, name, "meta")
```

Comment:
So in cases where there are some offloaded and some onloaded parameters, what happens?





[2:16](https://neuralmagic.slack.com/archives/C07MU79NX6U/p1734635792618969)
or is that not a case

Reply:
That case should not happen in practice. If that case does occur, then the newly registered parameter will remain onloaded.